### DISCUSSION TITLE
How to sign an Algorand transaction without posting it

### DISCUSSION THREAD

**Discussion by lorenzopolidori** [reach-sh/reach-lang#1462] [Q&A] [answered]
To implement a zero-algo / zero-fee transaction as an authentication token, I would need to sign a transaction with a connected wallet without posting it to the blockchain. 

What is the recommended approach using Reach? I see that once a wallet is connected, `window.algorand` exposes a `signTxns()` function. Is this the best way to do it?

**Comment by jeapostrophe** [comment, marked as answer]
I would recommend creating an object that obeys the ARC-11 wallet interface but wraps another instance of it. Reach will ask your "wallet" to sign, you'll ask the real wallet to sign, then you get your signed txn, then you just don't return it to Reach.
